### ISSUE TITLE
$ get best shell (sh, bash)

### ISSUE THREAD

**Issue by matti** [matti/kommando#62]

**Comment by SpComb** [comment]
The `Kommando.new('$ ...')` is [currently hardcoded to use `bash`](https://github.com/matti/kommando/blob/v0.1.0/lib/kommando.rb#L144):

```ruby
 ["bash", "-c", line]
```

 This fails if run in an environment which does not have `bash`, such as a Docker image based on `alpine`: https://github.com/kontena/kontena/issues/2468

Kommando should probably use `/bin/sh` instead of `bash` in this case?

---

[Using the kommando `$ ...` syntax](https://github.com/kontena/kontena/blob/v1.3.0/test/spec/features/service/exec_spec.rb#L47):

```ruby
    k = kommando("$ echo beer | kontena service exec -i test-1 rev")
```

Fails when running the spec in an alpine linux container:

```
  1) service exec runs a command with piped stdin
     Failure/Error: expect(k.run).to be_truthy
     
     Kommando::Error:
       Command 'bash' not found
     # ./spec/features/service/exec_spec.rb:48:in `block (2 levels) in <top (required)>'
     # ------------------
     # --- Caused by: ---
     # Errno::ENOENT:
     #   No such file or directory - fork failed
     #   ./spec/features/service/exec_spec.rb:48:in `block (2 levels) in <top (required)>'
```

Because the alpine linux base image does not include bash:

```
$ docker-compose run --rm test bash -v
ERROR: Cannot start service test: invalid header field value "oci runtime error: container_linux.go:247: starting container process caused \"exec: \\\"bash\\\": executable file not found in $PATH\"\n"
```

**Comment by matti** [comment]
should be fixed now with 0.1.1 ?

**Comment by matti** [comment]
fixed with 0.1.2
